jbpm-flow/src/main/java/org/jbpm/workflow/instance/node/EventSubProcessNodeInstance.java

```diff
@@ -54,14 +56,22 @@ public void signalEvent(String type, Object event) {
             // start it only if it was not already started - meaning there are node instances
             if (this.getNodeInstances().isEmpty()) {
                 StartNode startNode = getCompositeNode().findStartNode();
-                if (resolveVariables(((EventSubProcessNode) getEventBasedNode()).getEvents()).contains(type) || type.equals("timerTriggered")) {
+                if (resolveVariables(((EventSubProcessNode) getEventBasedNode()).getEvents()).contains(type) || containsTimerEvent(type, event)) {
                     NodeInstance nodeInstance = getNodeInstance(startNode);
                     ((StartNodeInstance) nodeInstance).signalEvent(type, event);
                 }
             }
         }
         super.signalEvent(type, event);
     }
+
+    private boolean containsTimerEvent(String type, Object event) {
+        if (!type.equals("timerTriggered") || !(event instanceof TimerInstance)) {
+            return false;
+        }
+        TimerInstance timerInstance = (TimerInstance) event;
+        return getTimerInstances().contains(timerInstance.getId());
```

Comment:
May `getTimerInstances` return null and this line throws a NPE?

Reply:
u r right. good catch